src/components/TodoContainer.js

```diff
@@ -0,0 +1,101 @@
+import React, { useState, useEffect } from 'react';
+import { Route, Switch } from 'react-router-dom';
+import { v4 as uuidv4 } from 'uuid';
+
+import Header from './Header';
+import TodoList from './TodoList';
+import InputTodo from './InputTodo';
+import Navbar from './Navbar';
+
+import About from '../page/About';
+import NotMatch from '../page/NotMatch';
+
+function getInitialTodos() {
+  // Get stored items from localstorage
+  const temp = localStorage.getItem('todos');
+  const savedTodos = JSON.parse(temp);
+  return savedTodos || [];
+}
+
+const TodoContainer = () => {
+  const [todos, setTodos] = useState(getInitialTodos());
+
```

Comment:
There is still a small issue that you will need to work on. The `getInitialTodos` function is not necessary when you use the `useEffect` Hook to perform that side effect(fetching data, directly updating the DOM...). The `useEffect` runs on every render. That means when the localStorage changes, a render happens, which trigger another effect and in this case, the `useEffect` hook will run the effect on the initial render, the reason why that function is not needed.
It would be better if you delete that function to keep it clean and YAGNI, and set the initial state to an empty array eg: `[state, setState] = useState([])`

Reply:
Thanks for the feedback. I will consider it.